encode.go

```diff
@@ -156,6 +156,27 @@ func (sm SortMode) valid() bool {
 	return sm >= 0 && sm < maxSortMode
 }
 
+// StringType is a CBOR string type.
+type StringType int
+
+const (
+	// TextStringType is the CBOR text string type (major type 3).
+	TextStringType StringType = iota
+
+	// ByteStringType is the CBOR byte string type (major type 2).
+	ByteStringType
+)
```

Comment:
Maybe use similar naming convention as existing options like `TimeMode`, etc.

```suggestion
// StringMode specifies how to encode Go string values.
type StringMode int

const (
	// StringToTextString encodes Go string to CBOR text string (major type 3).
	StringToTextString StringMode = iota

	// StringToByteString encodes Go string to CBOR byte string (major type 2).
	StringToByteString
)
```